### DISCUSSION TITLE
Getting error with terragrunt: Your current IAM principal doesn’t have access to Kubernetes objects on this cluster.

### DISCUSSION THREAD

**Discussion by rounakpr** [gruntwork-io/knowledge-base#723] [How do I]
Trying to launch EKS cluster with terragrunt. Cluster is Active now but i cannot see any worker nodes, although Node groups is available and worker nodes created. Error i am getting is : Your current IAM principal doesn’t have access to Kubernetes objects on this cluster.

Same is working fine if deployed with terraform alone. 

---

<ins datetime="2023-06-02T12:30:18Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110218">Tracked in ticket #110218</a></p>
</ins>

**Comment by pras111gw** [comment]
Hi @rounakpr are you using kubergrunt or eks-ctl to connect?

**Reply by rounakpr** [reply]
I am not able to verify the connection first, in the WEB UI i see the error `Your current IAM principal doesn’t have access to Kubernetes objects on this cluster`.
If the use the same script using terraform alone then every works fine.

Also when i check the Cloudtrail logs, i could see the `CreateCluster` event is being triggered by some random user (doesn't exist on AWS account). Not to forget, i also use the terragrunt command using the iam role `
$ terragrunt apply --terragrunt-iam-role arn:aws:iam::xxx:role/xxx`

**Reply by pras111gw** [reply]
Thanks. Do you mind sharing the following as well?
Terragrunt/Terraform versions
Gruntwork EKS module name and version
Is your IAM user an AWS admin? If not, could you share policies attached to it?

**Comment by rounakpr** [comment]
Was able to figure out the issue, it was because of IAM user missing permission

**Reply by vjwilson1987** [reply]
@rounakpr it would be nice if you share the workaround in detail here :)
